Src/AutoFixture/FreezeOnMatchCustomization.cs

```diff
@@ -75,6 +80,40 @@ public FreezeOnMatchCustomization(Type targetType)
         public IRequestSpecification Matcher { get; }
 
         /// <summary>
+        /// Used to create requests for resolving specimens before freezing.
+        /// </summary>
+        /// <remarks>
+        /// <para>
+        /// By default, the factory returns TargetType, but it can be tweaked 
+        /// to return any kind of request.
+        /// </para>
+        /// <para>
+        /// A typical use case is to create a request which includes significant 
+        /// information about a specimen. For instance, <see cref="ParameterInfo"/> 
+        /// which is widely used in Glue Libraries can contain custom attributes
+        /// such as <see cref="StringLengthAttribute"/> or <see cref="RegularExpressionAttribute" />.
+        /// These attributes are essential part of the request since they influence 
+        /// the final specimen.
+        /// </para>
+        /// </remarks>
+        public Func<object> RequestFactory
```

Comment:
Clearly, a factory's goal is to create something, but I don't understand why `FreezeOnMatchCustomization` needs to _create_ a new request every time it does what it has to do. Particularly not if the implementation always closes over a single object anyway...